### PR TITLE
Update Makefile

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Main changes:

* improve dependencies around variable filenames with timestamp file
* execute tests within venv

### Types of change

Bug fix.

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,36 +1,37 @@
...
 VENV := ./env$(PYVER)
 
-version = $(shell "bin/get-version.sh")
+version := $(shell "bin/get-version.sh")
 
-dist/spacy-$(version).pex : wheelhouse/spacy-$(version)-*.whl
-	pex -f ./wheelhouse --no-index --disable-cache -m spacy -o dist/spacy-$(version).pex spacy==$(version) jsonschema
-	chmod a+rx dist/spacy-$(version).pex
+dist/spacy-$(version).pex : wheelhouse/spacy-$(version).stamp
+	$(VENV)/bin/pex -f ./wheelhouse --no-index --disable-cache -m spacy -o $@ spacy==$(version) jsonschema
```
Comment (honnibal) [inline Makefile]: Ah not running pex within the virtualenv was was definitely an oversight. Nice idea with the timestamp too.

File: Makefile [hunk 2]
```diff
@@ -1,36 +1,37 @@
...
+$(VENV)/bin/pex :
 	python$(PYVER) -m venv $(VENV)
 	$(VENV)/bin/python -m pip install pex wheel
 
 .PHONY : clean test
 
 test : dist/spacy-$(version).pex dist/pytest.pex
-	PEX_PATH=dist/spacy-$(version).pex ./dist/pytest.pex --pyargs spacy -x
+	( . $(VENV)/bin/activate ; \
+	PEX_PATH=dist/spacy-$(version).pex ./dist/pytest.pex --pyargs spacy -x ; )
```
Comment (honnibal) [inline Makefile]: Why is this necessary? Pex shouldn't require the virtualenv to run I don't think? Is it just that it's picking the wrong Python for you?

Comment (adrianeboyd) [reply Makefile]: I think the problem is that `pex` isn't necessarily installed in the local venv? I got `ModuleNotFoundError: No module named 'pex'` otherwise.

File: Makefile (hunk 1 above)
Comment (adrianeboyd) [reply Makefile]: (Officemates who are better at Makefiles are helpful.)

**Comment by adrianeboyd** [comment]
One other thing that still isn't quite ideal is that `wheel` or `pip` doesn't seem to recythonize/rebuild correctly, so this is fine for a clean repo but didn't work well for me a normal working directory. Even `make clean; make` didn't seem to clean enough and rebuild enough. I had to rebuild separately with `python setup.py build_ext --inplace` and then it packaged the version I expected it to package.
